### PR TITLE
Allow customizing of the target binary

### PR DESCRIPTION
**Pull request by andreasmuellergin**
I myself stumbled over the issue of this not working for .net core executables. Below changes pose a solution for this. Currently i am using another solution but would be happy if you accepted this PR to allow me and others to use your package in .net core.

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
Are we sure that .NET Core even needs this change? This is only for executables. When the OS launches the .exe it reads the flag to determine whether to make the process large address aware. For dotnet, the dotnet.exe is the entrypoint process, and only the 32-bit one needs this change.

I can take this change but from my understanding it doesn't do anything. Or am I wrong?

**Comment by andreasmuellergin** [comment]
I'm a bit afraid i have a major misconception about how .net core executables work. 
My simplified setup goes like this ... 
If i create an netcoreapp3.1 app for x86 ...
```
<PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Platforms>x86</Platforms>
    <Platform>x86</Platform>
    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
  </PropertyGroup>
```

... it will create both a dll and an executable of this app. 
The dumpbin result of that executable will be ...


````
FILE HEADER VALUES
             14C machine (x86)
               5 number of sections
        625DBD1E time date stamp Mon Apr 18 21:33:50 2022
               0 file pointer to symbol table
               0 number of symbols
              E0 size of optional header
             102 characteristics
                   Executable
                   32 bit word machine
````
... missing the LAA flag, thus will make me want to set this flag for more memory.

Using the LLA nuget will set the flag for the dll. My proposed change allows me to set it for the executable instead.

**Comment by KirillOsenkov** [comment]
I see, OK. I understand and it makes sense. Surprised the .exe generated doesn’t have the flag set.

File: README.md [hunk 1]
```diff
@@ -22,3 +22,14 @@ You can customize after which target the SetLargeAddressAware target runs by set
 ```
 
 By default it runs after CoreCompile.
+
+You can also customize which target assembly should be modified by setting this property:
+
+```
+  <PropertyGroup>
+    <LargeAddressAwareTargetAssembly>bin/$(Configuration)/$(AssemblyName).exe</LargeAddressAwareTargetAssembly>
+    <LargeAddressAwareAfterTargets>AfterBuild</LargeAddressAwareAfterTargets>
```
Comment (KirillOsenkov) [inline README.md:31]: Don't see this used anywhere

**Review by KirillOsenkov — APPROVED** [review]

**Comment by KirillOsenkov** [comment]
Hmm, have you tried that it actually works? It doesn't seem to work for me because PlatformTarget is not set for SDK-style projects.

**Comment by KirillOsenkov** [comment]
For .NET 6 the apphost.exe is already marked with large address aware:

```
C:\>dumpbin /headers C:\temp\console\obj\Debug\net6.0\apphost.exe
Microsoft (R) COFF/PE Dumper Version 14.32.31328.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file C:\temp\console\obj\Debug\net6.0\apphost.exe

PE signature found

File Type: EXECUTABLE IMAGE

FILE HEADER VALUES
            8664 machine (x64)
               7 number of sections
        62571213 time date stamp Wed Apr 13 11:10:27 2022
               0 file pointer to symbol table
               0 number of symbols
              F0 size of optional header
              22 characteristics
                   Executable
                   Application can handle large (>2GB) addresses
```

**Comment by KirillOsenkov** [comment]
I have confirmed that the apphost for net5.0 and net6.0 is marked as Large Address Aware, so this is not needed for net5.0 and above. Not sure about 3.1.

**Comment by KirillOsenkov** [comment]
Hmm even for 3.1 I'm seeing the .exe automatically marked as LAA without this package.

How did you get the apphost .exe that doesn't have LAA?

**Comment by KirillOsenkov** [comment]
I have uploaded the version 1.0.6 of the package here:
https://www.nuget.org/packages/LargeAddressAware/1.0.6

You can try it and let me know if it works for you. As far as I can see it shouldn't be necessary for .NET Core but I can clearly see in your dumpbin that you do somehow need it.

**Comment by andreasmuellergin** [comment]
I am still uncertain how the final platform will be decided if a project platform selected in the VS2022 solution configuration manager or a platform in the csproj or platformTarget in the csproj contradict eachother. 

Using my config from above and selecting AnyCPU as project platform in the configuration manager will produce an application with x64, just as yours, also containing the LAA, regardless of the specified platform in the config.

My netcoreapp3.1 with the config above had x86 selected as project platform in the VS solution configuration manager. 
Thats how my output was different from yours. Selecting AnyCPU as project platform in the solution configuration manager but also setting ``<PlatformTarget>x86</PlatformTarger>`` in the config would lead to the same output like i posted earlier.


Using your released LAA 1.0.6 the output of dumpbin happily changes

```
FILE HEADER VALUES
             14C machine (x86)
               5 number of sections
        625DBD1E time date stamp Mon Apr 18 21:33:50 2022
               0 file pointer to symbol table
               0 number of symbols
              E0 size of optional header
             122 characteristics
                   Executable
                   Application can handle large (>2GB) addresses
                   32 bit word machine
```

when using

```
    <LargeAddressAware>true</LargeAddressAware>
    <LargeAddressAwareAfterTargets>AfterBuild</LargeAddressAwareAfterTargets> 
    <LargeAddressAwareTargetAssembly>bin/x86/$(Configuration)/$(AssemblyName).exe</LargeAddressAwareTargetAssembly>´
```

**Pull request merged**